google/cloud/storage/client_options.h

```diff
@@ -84,6 +84,17 @@ class ClientOptions {
     return *this;
   }
 
+  std::string const& user_agent_prefix() const { return user_agent_prefix_; }
+  ClientOptions& add_user_agent_prefx(std::string const& v) {
```

Comment:
Good question.  Partly because that is what gRPC does.  Partly for the same reason gRPC does it: people will build libraries layered on top of `google::cloud::storage`, and those libraries and the applications above them may want to add prefixes.